.github/workflows/py-workflow.yml

```diff
@@ -0,0 +1,348 @@
+        required: false
+        type: boolean
+        default: false
+        description: When 'true' black is run stopping the pipeline if any files are not properly formatted
+      with_docs:
+        required: false
+        type: boolean
+        default: false
+        description: When true documentation is published, the branch gh-pages must be created manually 'git checkout --orphan gh-pages && touch .nojekyll && git add .nojekyll && git commit -m "Initial commit" && git push --set-upstream origin gh-pages'
+      prod_publish_repos:
+        required: false
+        type: string
+        default: '["pypi-edge"]'
+    secrets:
+      auto_commit_user:
+        required: false # only when deploy true
+      auto_commit_mail:
+        required: false # only when deploy true
+      auto_commit_password:
+        required: false
+      registry_user:
+        required: true
+      registry_password:
+        required: true
+      nexus_publisher_user:
+        required: true
+      nexus_publisher_password:
+        required: true
+      gh_token:
+        required: true
+      sonar_token:
+        required: true
+
+jobs:
+  RefreshSource:
+    if: ${{ ! inputs.deploy }}
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: docker://chinthakagodawita/autoupdate-action:v1
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: '${{ secrets.auto_commit_password }}'
+          MERGE_MSG: "Branch was auto-updated."
+
+  Pre-commit:
+    if: ${{ inputs.pre_commit }}
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies and package
+        run: |
+          python3 -m pip install pip --upgrade
+          python3 -m pip install -i https://artifacts.cloud.mov.ai/repository/pypi-integration/simple --extra-index-url https://pypi.org/simple -r build-requirements.txt
+          python3 -m pip install -i https://artifacts.cloud.mov.ai/repository/pypi-integration/simple --extra-index-url https://pypi.org/simple .
+
+          python3 -m pip install pre-commit
+
+      - name: Run pre-commit
+        run: |
+          python3 -m pre_commit run --show-diff-on-failure --color=always --all-files
+
+  Build:
+    if: ${{ ! inputs.release }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.aws.cloud.mov.ai/devops/py-buildserver:latest
+      credentials:
+        username: ${{secrets.registry_user}}
+        password: ${{secrets.registry_password}}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Upgrade pip
+      run: python3 -m pip install pip --upgrade
+
+    - name: Install build-requirements
+      run: python3 -m pip install -r build-requirements.txt
+
+    - name: Install python package
+      run: python3 -m pip install .
+
+    - name: Run Black
```

Comment:
might be the right time to test ruff for replacement of black and flake8. see: https://github.com/astral-sh/ruff

Reply:
Maybe we can do this at a later date since formatting is only done when the input is set to true?